janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLKeyColumnValueStore.java

```diff
@@ -347,20 +348,41 @@ public String getName() {
 
     @Override
     public EntryList getSlice(final KeySliceQuery query, final StoreTransaction txh) throws BackendException {
+
+        if(executorService == null){
+            return getSliceWithoutExecutorService(query, txh);
+        }
+
+        return getSliceWithExecutorService(query, txh);
+    }
+
+    public EntryList getSliceWithExecutorService(final KeySliceQuery query, final StoreTransaction txh) throws BackendException {
         final Future<EntryList> result = Future.fromJavaFuture(
-                this.executorService,
-                this.session.executeAsync(this.getSlice.boundStatementBuilder()
-                        .setByteBuffer(KEY_BINDING, query.getKey().asByteBuffer())
-                        .setByteBuffer(SLICE_START_BINDING, query.getSliceStart().asByteBuffer())
-                        .setByteBuffer(SLICE_END_BINDING, query.getSliceEnd().asByteBuffer())
-                        .setInt(LIMIT_BINDING, query.getLimit())
-                        .setConsistencyLevel(getTransaction(txh).getReadConsistencyLevel()).build())
-                        .toCompletableFuture())
-                .map(resultSet -> fromResultSet(resultSet, this.getter));
+            this.executorService,
+            this.session.executeAsync(this.getSlice.boundStatementBuilder()
+                .setByteBuffer(KEY_BINDING, query.getKey().asByteBuffer())
+                .setByteBuffer(SLICE_START_BINDING, query.getSliceStart().asByteBuffer())
+                .setByteBuffer(SLICE_END_BINDING, query.getSliceEnd().asByteBuffer())
+                .setInt(LIMIT_BINDING, query.getLimit())
+                .setConsistencyLevel(getTransaction(txh).getReadConsistencyLevel()).build())
+                .toCompletableFuture())
+            .map(resultSet -> fromResultSet(resultSet, this.getter));
         interruptibleWait(result);
         return result.getValue().get().getOrElseThrow(EXCEPTION_MAPPER);
     }
 
+    public EntryList getSliceWithoutExecutorService(final KeySliceQuery query, final StoreTransaction txh) throws BackendException {
+        CompletableFuture<AsyncResultSet> asyncResult = this.session.executeAsync(this.getSlice.boundStatementBuilder()
+            .setByteBuffer(KEY_BINDING, query.getKey().asByteBuffer())
+            .setByteBuffer(SLICE_START_BINDING, query.getSliceStart().asByteBuffer())
+            .setByteBuffer(SLICE_END_BINDING, query.getSliceEnd().asByteBuffer())
+            .setInt(LIMIT_BINDING, query.getLimit())
+            .setConsistencyLevel(getTransaction(txh).getReadConsistencyLevel()).build())
+            .toCompletableFuture();
+        AsyncResultSet asyncResultSet = interruptibleWait(asyncResult);
+        return fromResultSet(asyncResultSet, this.getter);
+    }
```

Comment:
NIT: Most of the code is identical. I would put this code in a shared method and move the part that's different in `getSlice`. What do you think?